crates/runtime/src/init/dataset.rs

```diff
@@ -151,8 +150,14 @@ impl Runtime {
     }
 
     /// Returns a list of valid datasets from the given App, skipping any that fail to parse and logging an error for them.
-    pub(crate) fn get_valid_datasets(app: &Arc<App>, log_errors: LogErrors) -> Vec<Arc<Dataset>> {
-        Self::datasets_iter(app)
+    pub(crate) fn get_valid_datasets(
+        &self,
```

Comment:
```suggestion
        self: Arc<Self>,
```

We should just make this change an arc if we're going to need to clone it anyway.